tests/src/tracing/eventpipe/common/IpcTraceTest.cs

```diff
@@ -202,53 +191,83 @@ private int Validate()
             });
             sentinelTask.Start();
 
-            EventPipeEventSource source = null;
+            int processId = Process.GetCurrentProcess().Id;;
+            object threadSync = new object(); // for locking eventpipeSessionId access
+            ulong eventpipeSessionId = 0;
             Func<int> optionalTraceValidationCallback = null;
             var readerTask = new Task(() =>
             {
-                Logger.logger.Log("Creating EventPipeEventSource...");
-                source = new EventPipeEventSource(binaryReader);
-                Logger.logger.Log("EventPipeEventSource created");
-
-                source.Dynamic.All += (eventData) =>
+                Logger.logger.Log("Connecting to EventPipe...");
+                using (var eventPipeStream = new StreamProxy(EventPipeClient.CollectTracing(processId, _sessionConfiguration, out var sessionId)))
                 {
-                    try
+                    if (sessionId == 0)
+                    {
+                        Logger.logger.Log("Failed to connect to EventPipe!");
+                        throw new ApplicationException("Failed to connect to EventPipe");
+                    }
+                    Logger.logger.Log($"Connected to EventPipe with sessionID '0x{sessionId:x}'");
+
+                    lock(threadSync)
```

Comment:
nit: `lock (threadSync)` 

On a side note, why exactly is this lock needed? Is it because this can be accessed from the validation task and the task that sends IPC messages?

Reply:
The lock is because we write the value from inside the `readerTask` where we create the stream, but read it in a different thread for calling `StopTracing` while the other thread still exists.  I don't think these accesses will race ever due to the sentinel eventsource logic, but it's there as a sanity check.